### PR TITLE
New Okta related detection and hunting queries

### PR DESCRIPTION
**Pull request by aprakash13**
Required items, please complete
   
   Change(s):
   - New queries related to Okta

   Reason for Change(s):
   - New queries related to Okta

   Version Updated:
   -Yes

### CODE REVIEW

File: Solutions/Okta Single Sign-On/Analytic Rules/PhishingDetection.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+id: 78d2b06c-8dc0-40e1-91c8-66d916c186f3
+name: Okta Fast Pass phishing Detection
+description: |
+  'This query detects cases in which Okta FastPass effectively prevented access to a known phishing website'
+severity: High
```
Comment (petebryan) [inline Solutions/Okta Single Sign-On/Analytic Rules/PhishingDetection.yaml:5]: If this is blocked activity should it be high severity?

File: Solutions/Okta Single Sign-On/Hunting Queries/LegacyAuthentication.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ | where outcome_result_s == 'SUCCESS'
+ | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) by actor_alternateId_s, actor_displayName_s, client_userAgent_os_s, client_ipAddress_s, client_geographicalContext_state_s,displayMessage_s, outcome_result_s,eventType_s,
+ outcome_reason_s, column_ifexists('debugContext_debugData_logOnlySecurityData_s', ""), debugContext_debugData_threatSuspected_s, client_userAgent_rawUserAgent_s,client_userAgent_browser_s, severity_s,  client_geographicalContext_country_s, client_geographicalContext_city_s
+ | extend Account_0_Name = actor_displayName_s
+ | extend IP_0_Address = client_ipAddress_s
+entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: actor_displayName_s
```
Comment (petebryan) [inline Solutions/Okta Single Sign-On/Hunting Queries/LegacyAuthentication.yaml:27]: Is this a full UPN or just a username?

File: Solutions/Okta Single Sign-On/Hunting Queries/LoginFromMultipleLocations.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    | where End > Start
+    | project tostring(Start), tostring(End), locationString, EndLocationString, timeSpan = End - Start, actor_displayName_s, client_ipAddress_s, client_userAgent_rawUserAgent_s, client_userAgent_browser_s, client_device_s, displayMessage_s, outcome_result_s, outcome_reason_s, column_ifexists('debugContext_debugData_logOnlySecurityData_s', ""), debugContext_debugData_threatSuspected_s, client_geographicalContext_geolocation_lat_d, client_geographicalContext_geolocation_lon_d, eventType_s
+    | where locationString != EndLocationString
+    | summarize ips=make_set(client_ipAddress_s,100), UAs=make_set(client_userAgent_rawUserAgent_s,100) by timeSpan, actor_displayName_s, locationString, EndLocationString, Start, End, client_userAgent_rawUserAgent_s, client_userAgent_browser_s, client_device_s
+    | extend Account_0_Name = actor_displayName_s
+entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: actor_displayName_s
```
Comment (petebryan) [inline Solutions/Okta Single Sign-On/Hunting Queries/LoginFromMultipleLocations.yaml:62]: Is this a UPN?

File: Solutions/Okta Single Sign-On/Hunting Queries/LoginNewCountryDevice.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  | where parse_json(tostring(parse_json(debugContext_debugData_logOnlySecurityData_s).behaviors)).["New Geo-Location"] == "POSITIVE"
+  | where parse_json(tostring(parse_json(debugContext_debugData_logOnlySecurityData_s).behaviors)).["New Device"] == "POSITIVE"
+  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) by actor_alternateId_s, actor_displayName_s, client_userAgent_os_s, client_userAgent_browser_s, client_device_s, client_userAgent_rawUserAgent_s, client_ipAddress_s, client_geographicalContext_country_s, client_geographicalContext_city_s, displayMessage_s, outcome_result_s, outcome_reason_s, column_ifexists('debugContext_debugData_logOnlySecurityData_s', ""), debugContext_debugData_threatSuspected_s, client_geographicalContext_geolocation_lat_d, client_geographicalContext_geolocation_lon_d
+  | extend Account_0_Name = actor_displayName_s
+  | extend IP_0_Address = client_ipAddress_s
+entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: actor_displayName_s
```
Comment (petebryan) [inline Solutions/Okta Single Sign-On/Hunting Queries/LoginNewCountryDevice.yaml:29]: Is this a UPN?

File: Solutions/Okta Single Sign-On/Hunting Queries/LoginNordVPN.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) by actor_alternateId_s, actor_displayName_s, client_userAgent_os_s, client_ipAddress_s, client_geographicalContext_state_s,displayMessage_s, outcome_result_s,
+   outcome_reason_s, column_ifexists('debugContext_debugData_logOnlySecurityData_s', ""), debugContext_debugData_threatSuspected_s, client_userAgent_rawUserAgent_s,client_userAgent_browser_s, severity_s
+  | join kind= inner nord_vpn_feed on $left.client_ipAddress_s == $right.ip_address
+  | extend Account_0_Name = actor_displayName_s
+  | extend IP_0_Address = client_ipAddress_s
+entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: actor_displayName_s
```
Comment (petebryan) [inline Solutions/Okta Single Sign-On/Hunting Queries/LoginNordVPN.yaml:31]: Is this a UPN?

File: Solutions/Okta Single Sign-On/Hunting Queries/LoginsVPSProvider.yaml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  | evaluate ipv4_lookup(IP_Data, client_ipAddress_s, network, return_unmatched = false)
+  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) by actor_alternateId_s, actor_displayName_s, client_userAgent_os_s, client_ipAddress_s, client_geographicalContext_state_s,displayMessage_s, outcome_result_s,
+  outcome_reason_s, column_ifexists('debugContext_debugData_logOnlySecurityData_s', ""), debugContext_debugData_threatSuspected_s, client_userAgent_rawUserAgent_s,client_userAgent_browser_s, severity_s
+  | extend Account_0_Name = actor_displayName_s
+  | extend IP_0_Address = client_ipAddress_s
+entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: actor_displayName_s
```
Comment (petebryan) [inline Solutions/Okta Single Sign-On/Hunting Queries/LoginsVPSProvider.yaml:31]: Is this a UPN?

File: Solutions/Okta Single Sign-On/Analytic Rules/PhishingDetection.yaml (hunk 1 above)
Comment (aprakash13) [reply Solutions/Okta Single Sign-On/Analytic Rules/PhishingDetection.yaml]: Thanks Pete. Modified.

File: Solutions/Okta Single Sign-On/Hunting Queries/LegacyAuthentication.yaml (hunk 2 above)
Comment (aprakash13) [reply Solutions/Okta Single Sign-On/Hunting Queries/LegacyAuthentication.yaml:29]: When we think of UPN we normally think of username and domain combined with the @ separator. In the Okta logs I didn't see the identity for most events in the UPN format. So, I have added two identifiers for the Account entity. A **Name** which can be though of as the unique name of an account similar to our alias and a **DisplayName** that is the complete name of the account.

File: Solutions/Okta Single Sign-On/Hunting Queries/LoginFromMultipleLocations.yaml (hunk 3 above)
Comment (aprakash13) [reply Solutions/Okta Single Sign-On/Hunting Queries/LoginFromMultipleLocations.yaml:64]: Same as previous UPN comment

File: Solutions/Okta Single Sign-On/Hunting Queries/LoginNewCountryDevice.yaml (hunk 4 above)
Comment (aprakash13) [reply Solutions/Okta Single Sign-On/Hunting Queries/LoginNewCountryDevice.yaml]: Same as previous UPN comment

File: Solutions/Okta Single Sign-On/Hunting Queries/LoginsVPSProvider.yaml (hunk 6 above)
Comment (aprakash13) [reply Solutions/Okta Single Sign-On/Hunting Queries/LoginsVPSProvider.yaml:33]: When we think of UPN we normally think of username and domain combined with the @ separator. In the Okta logs I didn't see the identity for most events in the UPN format. So, I have added two identifiers for the Account entity. A Name which can be though of as the unique name of an account similar to our alias and a display name that is the complete name of the account.

File: Solutions/Okta Single Sign-On/Hunting Queries/LoginNordVPN.yaml (hunk 5 above)
Comment (aprakash13) [reply Solutions/Okta Single Sign-On/Hunting Queries/LoginNordVPN.yaml:33]: When we think of UPN we normally think of username and domain combined with the @ separator. In the Okta logs I didn't see the identity for most events in the UPN format. So, I have added two identifiers for the Account entity. A Name which can be though of as the unique name of an account similar to our alias and a display name that is the complete name of the account.

**Review by petebryan — COMMENTED** [review]
These are great @aprakash13 ! I love the use cases here, some really good detection opportunities. I've added some comments where we could maybe project out a few other things from the query to make them more efficient to triage.

File: Solutions/Okta Single Sign-On/Analytic Rules/NewDeviceLocationCriticalOperation.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  let RiskyOperationsObserved = Okta_CL
+  | where eventType_s in (RiskyOperations)
+  | where outcome_result_s == "SUCCESS"
+  | summarize by timekey = bin(TimeGenerated, timeframe), actor_alternateId_s, actor_displayName_s, client_userAgent_os_s, client_userAgent_browser_s, client_device_s, client_userAgent_rawUserAgent_s, client_ipAddress_s, authenticationContext_externalSessionId_s, client_geographicalContext_country_s, client_geographicalContext_city_s, displayMessage_s, outcome_result_s, outcome_reason_s, column_ifexists('debugContext_debugData_logOnlySecurityData_s', ""), debugContext_debugData_threatSuspected_s, client_geographicalContext_geolocation_lat_d, client_geographicalContext_geolocation_lon_d;
```
[the remark this answers is not in the document]
Comment (aprakash13) [reply Solutions/Okta Single Sign-On/Analytic Rules/NewDeviceLocationCriticalOperation.yaml:35]: Thanks Pete. Addressed.

File: Solutions/Okta Single Sign-On/Analytic Rules/MFAFatigue.yaml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+            pushes = countif(eventType_s == "system.push.send_factor_verify_push") by TimeGenerated, authenticationContext_externalSessionId_s, actor_alternateId_s,actor_displayName_s, outcome_result_s 
+  | summarize lasttime = max(TimeGenerated), firsttime = min(TimeGenerated),
+            successes = sum(successes), failures = sum(denies), pushes = sum(pushes) by  authenticationContext_externalSessionId_s, actor_alternateId_s,actor_displayName_s, outcome_result_s 
+  | extend seconds = lasttime - firsttime
+  | where pushes >  (PushThreshold)
+  | extend totalattempts = successes + failures
+  | extend finding = case(
+              failures == pushes and pushes > 1, "Authentication attempts not successful because multiple pushes denied",
+              totalattempts == 0, "Multiple pushes sent and ignored",
+              successes > 0 and pushes > 3, "Multiple pushes sent, eventual successful authentication!",
```
[the remark this answers is not in the document]
Comment (aprakash13) [reply Solutions/Okta Single Sign-On/Analytic Rules/MFAFatigue.yaml:36]: Will address in the next PR along with Session cookie detection.

**Review by petebryan — APPROVED** [review]

**Pull request merged**